src/pages/index.js

```diff
@@ -0,0 +1,375 @@
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      editProfileSubmitButton.textContent = "Submit";
+    });
+}
+
+function editAvatar(evt) {
+  evt.preventDefault();
+  api
+    .editUserAvatar({
+      avatar: editAvatarInput.value,
+    })
+    .then((data) => {
+      avatarImg.src = data.avatar;
+      editAvatarInput.value = "";
+      closeModal(editAvatarModal);
+    })
+    .catch(console.error);
+}
+
+function getCardElement(data) {
+  const cardTemplate = document.querySelector("#card-template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const likeButton = cardElement.querySelector(".card__like-btn");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Set the initial liked status
+  if (data.isLiked) {
+    likeButton.classList.add("card__like-btn_liked");
+  }
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(viewImageModal);
+    if (modalImage && modalCaption) {
+      modalImage.src = data.link;
+      modalImage.alt = data.name;
+      modalCaption.textContent = data.name;
+    }
+  });
+
+  likeButton.addEventListener("click", () => {
+    if (likeButton.classList.contains("card__like-btn_liked")) {
+      api
+        .unlikeCard(data._id)
+        .then(() => {
+          likeButton.classList.remove("card__like-btn_liked");
+          data.isLiked = false;
+        })
+        .catch((error) => {
+          console.error(error);
+        });
+    } else {
+      api
+        .likeCard(data._id)
+        .then(() => {
+          likeButton.classList.add("card__like-btn_liked");
+          data.isLiked = true;
+        })
+        .catch((error) => {
+          console.error(error);
+        });
+    }
+  });
+
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+  deleteButton.addEventListener("click", () => {
+    const cardToDelete = cardElement;
+
+    openModal(deleteImageModal);
+
+    deleteImageDeleteButton.onclick = () => {
+      // Change the button text to "Deleting..."
+      deleteImageDeleteButton.textContent = "Deleting...";
+
+      api
+        .deleteCard(data._id)
+        .then(() => {
+          cardToDelete.remove();
+          closeModal(deleteImageModal);
+        })
+        .catch((error) => {
+          console.error(error);
+        })
+        .finally(() => {
+          // Change the button text back to "Delete"
+          deleteImageDeleteButton.textContent = "Delete";
+        });
+    };
+
+    deleteImageCancelButton.onclick = () => {
+      closeModal(deleteImageModal);
+    };
```

Comment:
No need in adding the close modal listener every time you open the confirmation modal. It should be done only **once** outside `getCardElement`